corehq/project_limits/rate_limiter.py

```diff
@@ -158,12 +175,25 @@ def __init__(self, per_user_rate_definition, constant_rate_definition=None):
         self.constant_rate_definition = constant_rate_definition or RateDefinition()
 
     def get_rate_limits(self, domain):
-        n_users = get_n_users_for_rate_limiting(domain)
-        return (
-            self.per_user_rate_definition
-            .times(n_users)
-            .plus(self.constant_rate_definition)
-        ).get_rate_limits()
+        domain_users = get_n_users_in_domain(domain)
+        enterprise_users = get_n_users_in_subscription(domain)
+        limit_pairs = [
+            (domain_users, domain),
+            (enterprise_users, _get_account_name(domain))
```

Comment:
My read here is that all `scope_key`s share a namespace, but that domain names and enterprise account names do _not_ currently share a namespace, i.e. you can have a domain and account that have the same name (whether they're related or not), thus causing collisions where using that `scope_key` ambiguously refers to either the domain or the account. Could we fix this by renaming `_get_account_name` to `get_acount_scope_key` and having it return `f'account:{account.name}'`?

Not sure if the domain scope key needs to match up perfectly with existing redis keys or something, but if it doesn't then it might be worth prefixing that one as well (`f'domain:{domain}'`). If it does, then probably not worth the change.

Reply:
That is a good idea for account, i think domain has the issue you mentioned, but as long as the other ones are prefixed that should be fine.